fcrepo4/fcrepo4
# Eliminate stacktrace when accessing /fcr:metadata

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2867

# What does this Pull Request do?
Eliminate stacktrace when accessing /fcr:metadata

# What's new?
Remove cast to FedoraBinaryImpl since getEffectiveAcl already handles detecting binary nodes gracefully.

# How should this be tested?
Follow the instructions in the original ticket. Observe that when doing a GET or HEAD request to the /fcr:metadata resource, there is a 200 OK response and no stacktrace.

# Interested parties
@whikloj , @fcrepo4/committers
